packages/contracts/contracts/libraries/Hashing.sol

```diff
@@ -198,7 +198,37 @@ library Hashing {
     }
 
     /**
-     * @notice Hashes the various elements of a block header into a block hash.
+     * @notice Fills the values of the block hash fields to a given bytes.
+     *
+     * @param _publicInput Public input which should hash to a block hash.
+     * @param _rlps        Pre-RLP encoded data which should hash to a block hash.
+     * @param _raw         An array of bytes to be populated.
+     */
+    function _fillBlockHashFieldsToBytes(
+        Types.PublicInput memory _publicInput,
+        Types.BlockHeaderRLP memory _rlps,
+        bytes[] memory _raw
+    ) private pure {
+        _raw[0] = RLPWriter.writeBytes(abi.encodePacked(_publicInput.parentHash));
+        _raw[1] = _rlps.uncleHash;
+        _raw[2] = _rlps.coinbase;
+        _raw[3] = RLPWriter.writeBytes(abi.encodePacked(_publicInput.stateRoot));
+        _raw[4] = RLPWriter.writeBytes(abi.encodePacked(_publicInput.transactionsRoot));
+        _raw[5] = _rlps.receiptsRoot;
+        _raw[6] = _rlps.logsBloom;
+        _raw[7] = _rlps.difficulty;
+        _raw[8] = RLPWriter.writeUint(_publicInput.number);
+        _raw[9] = RLPWriter.writeUint(_publicInput.gasLimit);
+        _raw[10] = _rlps.gasUsed;
+        _raw[11] = RLPWriter.writeUint(_publicInput.timestamp);
+        _raw[12] = _rlps.extraData;
+        _raw[13] = _rlps.mixHash;
+        _raw[14] = _rlps.nonce;
+        _raw[15] = RLPWriter.writeUint(_publicInput.baseFee);
```

Comment:
The order is changed as well. Is it required changes for generating proof?
I'm not familiar with the proof and public info things, but it would be good to give some hints for this change.

Reply:
Please see https://github.com/scroll-tech/zkevm-circuits/blob/v0.3.0-hotfix1/zkevm-circuits/src/pi_circuit.rs#L101-L179

Reply:
Here `num_l1_msgs` are removed as we don't need this.